### PR TITLE
Consolidate duplicated logic on reset tools

### PR DESCRIPTION
**Pull request by manijndl7**
**Consolidated Duplicate DateTime logic from ConsumerGroupCommand and StreamResetter in common Client Utils**

[https://issues.apache.org/jira/browse/KAFKA-6585](url)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
@brary Wanna help reviewing this PR?

**Comment by brary** [comment]
> @brary Wanna help reviewing this PR?

Sure @mjsax, would be happy to pitch in.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 1]
```diff
@@ -16,10 +16,6 @@
  */
 package org.apache.kafka.common.utils;
 
-import java.nio.BufferUnderflowException;
```
Comment (brary) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java:19]: @manijndl7 Please revert any unnecessary changes in imports. The imports removed are added again at the bottom. Please add or remove only the diff.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 2]
```diff
@@ -1271,4 +1274,28 @@ private static byte checkRange(final byte i) {
         }
         return map;
     }
+
+    public static long getDateTime(String timestamp) throws ParseException {
```
Comment (brary) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java:1278]: Since this is a public function, it would be good to add a javadoc. Please refer to the function above in the same class for example.

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala [hunk 3]
```diff
@@ -17,34 +17,29 @@
...
 import kafka.utils._
+import org.apache.kafka.clients.CommonClientConfigs
 import org.apache.kafka.clients.admin._
 import org.apache.kafka.clients.consumer.OffsetAndMetadata
-import org.apache.kafka.clients.CommonClientConfigs
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.requests.ListOffsetResponse
 import org.apache.kafka.common.utils.Utils
-import org.apache.kafka.common.{KafkaException, Node, TopicPartition}
+import org.apache.kafka.common.{ConsumerGroupState, KafkaException, Node, TopicPartition}
```
Comment (brary) [inline core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:34]: same comment as above, please revert any unnecessary changes in imports.

File: core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala [hunk 4]
```diff
@@ -26,43 +26,10 @@ import org.apache.kafka.test
 import org.junit.Assert._
 import org.junit.Test
 
-import scala.jdk.CollectionConverters._
```
Comment (brary) [inline core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala:29]: same as above.

**Review by brary — COMMENTED** [review]
Thanks for the PR Mani, overall looks good to me apart from the nitpicks. The test failures look unrelated as they have failed in other recently closed PRs as well.

File: core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala (hunk 4 above)
Comment (manijndl7) [reply core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala:29]: @brary  Hi Navi They are indent automatically by Intellij can you please help me with Intellij Code Style XML so that i will indent it accordingly.

Comment (brary) [reply core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala:29]: You can manually cut/paste the changed imports back. It will revert them to the original commit.

**Comment by manijndl7** [comment]
Closing This Created New PR : https://github.com/apache/kafka/pull/9255
